File: app/src/main/java/br/com/iomarsantos/testeandroid/di/component/ApplicationComponent.java
package br.com.iomarsantos.testeandroid.di.component;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import br.com.iomarsantos.testeandroid.TesteAndroidApp;
import br.com.iomarsantos.testeandroid.data.Repository;
import br.com.iomarsantos.testeandroid.di.ApplicationContext;
import br.com.iomarsantos.testeandroid.di.module.ApplicationModule;
import dagger.Component;

@Singleton
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {

    void inject(TesteAndroidApp testeAndroidApp);

    @ApplicationContext
    Context context();

    Application application();

    Repository getRepository();

}

File: app/src/main/java/andcordeiro/com/br/testeandroid/system/dagger/ApplicationComponent.kt
package andcordeiro.com.br.testeandroid.system.dagger

import andcordeiro.com.br.testeandroid.histories.contact.ContactFragment
import andcordeiro.com.br.testeandroid.histories.contact.ContactModule
import andcordeiro.com.br.testeandroid.histories.investiment.InvestimentFragment
import andcordeiro.com.br.testeandroid.histories.investiment.InvestimentModule
import andcordeiro.com.br.testeandroid.system.retrofit.RetrofitApiModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ApplicationModule::class, ContactModule::class,
        InvestimentModule::class, RetrofitApiModule::class))
interface ApplicationComponent {

    fun inject(target: ContactFragment)

    fun inject(target: InvestimentFragment)
}